ddtrace/propagation/http.py

```diff
@@ -30,7 +30,9 @@ def parent_call():
         """
         headers[HTTP_HEADER_TRACE_ID] = str(span_context.trace_id)
         headers[HTTP_HEADER_PARENT_ID] = str(span_context.span_id)
-        headers[HTTP_HEADER_SAMPLING_PRIORITY] = str(span_context.sampling_priority)
+        sampling_priority = span_context.sampling_priority
+        if sampling_priority is not None: # avoid converting None to 'None'
```

Comment:
Nasty bug I found when writing tests, this is currently broken, any server implementation not handling a random string (`'None'` here) could end up with an exception.

Reply:
Do we need a patch to other libraries to ensure that we never `int('None')`?

Reply:
I think we should not patch any other library but instead ensure all libraries use this shared piece of code. As of today, any library using this features the bug, we should fix them all by releasing this. A quick `grep` shows `x-datadog-sampling-priority` is only set by this call so we should be fine by fixing just this piece of code.

Reply:
Perfect! 👍